custom_components/adjustable_bed/binary_sensor.py

```
@@ -0,0 +1,123 @@
+        self._unregister_callback = self._coordinator.register_connection_state_callback(
+            self._handle_connection_state_change
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity is removed from hass."""
+        if self._unregister_callback:
+            self._unregister_callback()
+        await super().async_will_remove_from_hass()
+
+    @callback
+    def _handle_connection_state_change(self, connected: bool) -> None:
```

Ruff (0.14.13)

Boolean-typed positional argument in function definition

(FBT001)